pkg/controller/syncmachineset/syncmachineset_controller.go

```diff
@@ -0,0 +1,555 @@
+	coCluster, err := c.client.ClusteroperatorV1alpha1().Clusters(ms.Namespace).Get(ms.Labels[controller.ClusterNameLabel], metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error looking up machineset's cluster: %v", err)
+	}
+	coMasterMachineSet, err := c.client.ClusteroperatorV1alpha1().MachineSets(coCluster.Namespace).Get(coCluster.Status.MasterMachineSetName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error looking up cluster's master machineset: %v", err)
+	}
+
+	// handleErr will requeue machine sets after waiting when the
+	// remote cluster API is not installed.
+	if !coMasterMachineSet.Status.ClusterAPIInstalled {
+		return NewErrClusterAPINotInstalled(errorMsgClusterAPINotInstalled)
+	}
+
+	remoteClusterAPIClient, remoteDeploymentClient, err := c.buildClients(ms)
+	if err != nil {
+		return err
+	}
+
+	// handleErr will requeue machine sets after waiting when the
+	// remote cluster API deployment is not ready.
+	remoteClusterAPIDeployment, err := remoteDeploymentClient.AppsV1().Deployments(remoteClusterAPINamespace).Get(remoteClusterAPIDeploymentName, metav1.GetOptions{})
+	if err != nil {
+		return NewErrClusterAPINotInstalled(fmt.Sprintf("unable to retrieve remote cluster API deployment: %v", err))
+	}
+	if remoteClusterAPIDeployment.Status.ReadyReplicas < 1 {
+		return NewErrClusterAPINotInstalled(errorMsgClusterAPINotInstalled)
+	}
+
+	// Create machineSet object in remote if the machineSet does not already exist.
+	remoteMachineSet, err := remoteClusterAPIClient.ClusterV1alpha1().MachineSets(remoteClusterAPINamespace).Get(ms.Name, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		msLog.Infof("creating new machineSet API object in remote cluster")
+
+		// Lookup clusterVersion for machineSet
+		coVersionNamespace, coVersionName := ms.Spec.ClusterVersionRef.Namespace, ms.Spec.ClusterVersionRef.Name
+		coClusterVersion, err := c.client.ClusteroperatorV1alpha1().ClusterVersions(coVersionNamespace).Get(coVersionName, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("cannot retrieve cluster version %s/%s: %v", coVersionNamespace, coVersionName, err)
+		}
+
+		remoteMachineSet = buildClusterAPIMachineSet(ms, coClusterVersion)
+		_, erree := remoteClusterAPIClient.ClusterV1alpha1().MachineSets(remoteClusterAPINamespace).Create(remoteMachineSet)
+		if erree != nil {
+			return fmt.Errorf("error creating machineSet API object in remote cluster: %v", erree)
+		}
+	} else if err != nil {
+		return err
+	} else {
+		// TODO: patch/update here?
+		msLog.Debugf("machineSet API object already exists in remote cluster")
+	}
+
+	return nil
+}
+
+func buildClusterAPIMachineSet(machineSet *cov1.MachineSet, clusterVersion *cov1.ClusterVersion) *clusterapiv1.MachineSet {
+	if machineSet.Annotations == nil {
+		machineSet.Annotations = map[string]string{}
+	}
+	machineSet.Annotations["cluster-operator.openshift.io/cluster-version"] = serializeCOResource(clusterVersion)
```

Comment:
Move this down to where `machineSet` is serialized into the machine template. Here it looks like a mistake that the annotation is added to `machineSet` instead of `capiMachineSet`.

Also, do a deep copy of `machineSet` and use that instead to be clear that you are not intending to modify the original machine set.